src/SFA.DAS.ApprenticeCommitments.Api/Controllers/ApprenticeshipController.cs

```diff
@@ -19,5 +20,13 @@ public async Task<IActionResult> AddApprenticeship(CreateApprenticeshipCommand r
             await _mediator.Send(request);
             return Accepted();
         }
+
+        [HttpPost("/apprentices/{apprenticeId}/email")]
+        public async Task<IActionResult> ChangeApprenticeEmailAddress(long apprenticeId, ChangeEmailAddressCommand request)
```

Comment:
Naming is hard :)

I am generally of the opinion that it *is* a command - the client is telling the API to change the email address.  I find "Request" to be a bit vague - is it a request for data, a request for some action to be performed, ...?

The command/request is transmitted to the API via a HTTP Request, but that term Request is really nothing to do with the application.

I used the term "command" here mostly because it is a Mediatr `Command`, and it's being used for the API surface as well.  I generally dislike having hundreds of the-same-but-with-a-different-namespace classes to represent the same data as it travels through layers.

However, in this case there is actually a potential benefit of having a separate class for the API contract - the Apprentice ID comes from the querystring, *not* from the body.  In fact, any ID sent as part of the body will be ignored.  Note this is still true if we remove the ID property from the HTTP API object - a client could still send a body with `ApprenticeId` and it would still be ignored, although that would simply be extra unexpected data.  Having the property visible in the API class does lend some incorrect legitimacy to the idea that the ID could be set in the body.

To change that we would have to create a new API DTO class that is the same as the Mediatr command minus the ID.

Alternatively we could remove the ID from the querystring, straying further from the SFA guidance "[you should build your API around RESTful principles](https://skillsfundingagency.github.io/das-technical-guidance/api_standards/2-api-design#build-a-restful-api)."